docs/troubleshooting.md

```diff
@@ -0,0 +1,90 @@
+# Common troubleshooting tips
+
+## Module not found errors
+
+This is the most common error rules_js users encounter.
+These problems generally stem from a runtime `require` call of some library which was not declared as a dependency.
+
+Fortunately, these problems are not unique to Bazel.
+As described in https://github.com/aspect-build/rules_js/blob/main/docs/pnpm.md#hoisting, rules_js
+should behave the same way `pnpm` does with `hoist=false`.
+These problems are also reproducible under [Yarn PnP](https://yarnpkg.com/features/pnp) because it
+also relies on correct dependencies.
+
+The Node.js documentation describes the algorithm used:
+https://nodejs.org/api/modules.html#loading-from-node_modules-folders
+
+Since the resolution starts from the callsite, the remedy depends on where the `require` statement appears.
+
+### require appears in your code
+
+This is the case when you write a `config.js` file and pass it to a tool.
+
+> This is the "ideal" way for JavaScript libraries to be configured, because it allows an easy
+> "symmetry" where you `require` a library and declare your dependency on it in the same package.
```

Comment:
"package" is a bit of an overloaded word, in this case we mean bazel target? The `srcs` and `deps of the same target...